invokeai/app/services/workflow_records/workflow_records_sqlite.py

```diff
@@ -24,76 +29,286 @@ def __init__(self, db: SqliteDatabase) -> None:
+                    updated_at DATETIME NOT NULL DEFAULT(STRFTIME('%Y-%m-%d %H:%M:%f', 'NOW')),
+                    -- updated manually when retrieving workflow
+                    opened_at DATETIME NOT NULL DEFAULT(STRFTIME('%Y-%m-%d %H:%M:%f', 'NOW')),
+                    -- Generated columns, needed for indexing and searching
+                    category TEXT GENERATED ALWAYS as (json_extract(workflow, '$.meta.category')) VIRTUAL NOT NULL,
+                    name TEXT GENERATED ALWAYS as (json_extract(workflow, '$.name')) VIRTUAL NOT NULL,
+                    description TEXT GENERATED ALWAYS as (json_extract(workflow, '$.description')) VIRTUAL NOT NULL
                 );
                 """
             )
 
             self._cursor.execute(
                 """--sql
-                CREATE TRIGGER IF NOT EXISTS tg_workflows_updated_at
+                CREATE TRIGGER IF NOT EXISTS tg_workflow_library_updated_at
                 AFTER UPDATE
-                ON workflows FOR EACH ROW
+                ON workflow_library FOR EACH ROW
                 BEGIN
-                    UPDATE workflows
+                    UPDATE workflow_library
                     SET updated_at = STRFTIME('%Y-%m-%d %H:%M:%f', 'NOW')
                     WHERE workflow_id = old.workflow_id;
                 END;
                 """
             )
 
+            self._cursor.execute(
+                """--sql
+                CREATE INDEX IF NOT EXISTS idx_workflow_library_created_at ON workflow_library(created_at);
+                """
+            )
+            self._cursor.execute(
+                """--sql
+                CREATE INDEX IF NOT EXISTS idx_workflow_library_updated_at ON workflow_library(updated_at);
+                """
+            )
+            self._cursor.execute(
+                """--sql
+                CREATE INDEX IF NOT EXISTS idx_workflow_library_opened_at ON workflow_library(opened_at);
+                """
+            )
+            self._cursor.execute(
+                """--sql
+                CREATE INDEX IF NOT EXISTS idx_workflow_library_category ON workflow_library(category);
+                """
+            )
+            self._cursor.execute(
+                """--sql
+                CREATE INDEX IF NOT EXISTS idx_workflow_library_name ON workflow_library(name);
+                """
+            )
+            self._cursor.execute(
+                """--sql
+                CREATE INDEX IF NOT EXISTS idx_workflow_library_description ON workflow_library(description);
+                """
+            )
+
+            # We do not need the original `workflows` table or `workflow_images` junction table.
+            self._cursor.execute(
+                """--sql
+                DROP TABLE IF EXISTS workflow_images;
+                """
+            )
+            self._cursor.execute(
+                """--sql
+                DROP TABLE IF EXISTS workflows;
+                """
+            )
```

Comment:
Instead of doing a data migration on the `workflows` table to make it into the workflow library, we perma-ban both `workflows` and `workflow_images` 😅